core/sync.js

```diff
@@ -123,23 +135,30 @@ class Sync {
       this.stop()
       throw err
     })
+    log.info('Sync started')
   }
 
   // Stop the synchronization
-  stop() {
-    this.stopped = true
-    if (this.changes) {
-      this.changes.cancel()
-      this.changes = null
+  async stop() {
+    log.info('Stopping Sync...')
+    if (this.runningResolve) {
+      this.runningResolve()
+      this.runningResolve = null
     }
-    return Promise.all([this.local.stop(), this.remote.stop()])
+    await this.stopped
+    log.info('Sync stopped')
```

Comment:
I don't know if it is an issue, but if there was a change being applied, `stop` can return before it has finished, and when the change will finish, it will write to PouchDB. Maybe we should look if we can cancel the change.

Reply:
It would be better to wait until the change is applied. The goal of the PR is to properly stop the synchronization so it should be part of it.

Reply:
Even if it is a long download?

Reply:
Since we don't rollback changes saved in Pouch but not completely applied, yes we should.

Reply:
Does it mean that the client won't exit when the user asks to quit until the download as finished?

Reply:
I'll run a test to see what happens.

Reply:
I've ran a test and we do wait for the end of the download before exiting the client.
I think it's a good thing knowing how we manage inconsistencies in Pouch today but I understand this might be upsetting to some users :/.

Reply:
I don't follow you. As I understand, the sync doesn't save anything in PouchDB until the download has finished. And if we stop the download before its end, we can find a trick to not write at all in PouchDB for this operation. So, we will have no inconsistencies.

Reply:
Indeed the Sync won't write in PouchDB until the download is done.
But the Merge has already written in PouchDB the new version of the document. If by any chance this was not a new document but an update, we'll end up loosing the update by overwriting the remote version with out-of-date local one (on top of a conflict reanming :/).

Reply:
Can we find another way to fix this issue? Because, if I understand correctly, if several files are updated on the Cozy, not stopping the download will only save the first one, the two other will still be corrupted when cozy-desktop will be restarted.

Reply:
You're right, this won't help with multiple downloads. This wasn't mean as a fix for this issue though.

Reply:
I've introduced a specific method for the manual sync that will wait for the current sync operation to end while leaving the app won't wait.